pkgs/by-name/ra/ratchet/package.nix

```
@@ -0,0 +1,38 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
+}: buildGoModule rec {
+  pname = "ratchet";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "sethvargo";
+    repo = "ratchet";
+    rev = "v${version}";
+    hash = "sha256-xN/c9QKnxf8bxciAOrP0ZIpuGRbIYM6Gs2lvbXQAniA=";
+  };
+
+  vendorHash = "sha256-RDa2AlZlZdthLlocG4E1TcgeQmjlArpMrUzhQhfvioA=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=github.com/sethvargo/ratchet/internal/version.name=ratchet"
+    "-X=github.com/sethvargo/ratchet/internal/version.version=${version}"
+    "-X=github.com/sethvargo/ratchet/internal/version.commit=${src.rev}"
+  ];
+
+  # skip tests that require network access
+  checkFlags =
+    let
+      skippedTests = [ "TestResolve" ];
+    in
+    [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
```

i don't think we need to be this careful with the regex, plus using two separate strings makes things a bit neater. the comment about network access should also only apply to the test it's referring to, not any test that may be skipped

```suggestion
  checkFlags =
    let
      skippedTests = [
        "TestResolve" # requires network access
      ];
    in
    [ "-skip" (lib.concatStringsSep "|" skippedTests) ];
```